src/services/notifications/push-notifications.ts

```diff
@@ -112,7 +115,40 @@ export const PushNotificationsService = (): IPushNotificationsService => {
     })()
   }
 
-  return { sendNotification }
+  const sendFilteredNotification = async (args: SendFilteredPushNotificationArgs) => {
+    const { notificationSettings, notificationCategory, data, ...sendNotificationArgs } =
+      args
+
+    if (
+      !shouldSendNotification({
+        notificationCategory,
+        notificationSettings,
+        notificationChannel: NotificationChannel.Push,
+      })
+    ) {
+      return {
+        status: SendFilteredPushNotificationStatus.Filtered,
+      }
+    }
+
+    const result = await sendNotification({
+      ...sendNotificationArgs,
+      data: {
+        ...data,
+        NotificationCategory: notificationCategory,
```

Comment:
not sure what is supposed to be inside { data } versus at the root level (ie: `{ ...sendNotification, ... }`) but reading here would make me think  `NotificationCategory: notificationCategory,` is more a metadata thing and should be one level up in the root

Reply:
That makes sense, I will move it up a level.

Reply:
Actually I think it may make sense to keep it where it is at. It is preferable that all notifications that have a notification category receive it as a part of their metadata. Since this method already has logic relating to handling notification categories I think it is alright for it to augment the metadata.